test/unit/directivesSpec.js

```diff
@@ -157,4 +157,184 @@ describe('directives', function() {
+      expect($scope.Form.$valid).toBeUndefined(true);
+      expect($scope.Form.$invalid).toBeUndefined(true);
+    });
+
+    it('should be dirty and valid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+      $scope.Form.telephone.$setViewValue('065839481');
+
+      expect($scope.Form.$dirty).toBe(true);
+      expect(element.hasClass('ng-dirty')).toBe(true);
+      expect($scope.Form.$valid).toBe(true);
+      expect(element.hasClass('ng-valid')).toBe(true);
+    });
+
+    it('should be dirty and invalid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+      $scope.Form.telephone.$setViewValue('065839481');
+      $scope.Form.email.$setViewValue();
+      $scope.Form.telephone.$setViewValue();
+
+      expect($scope.Form.$dirty).toBe(true);
+      expect(element.hasClass('ng-dirty')).toBe(true);
+      expect($scope.Form.$invalid).toBe(true);
+      expect(element.hasClass('ng-invalid')).toBe(true);
+    });
+
+    it('should have a success message inside the #contact element when an element is valid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+
+      messageElem = angular.element(element[0].querySelector('#contact > p'));
+      expect(messageElem.hasClass('validation-valid')).toBe(true);
+    });
+
+    it('should have an error message inside the #contact element when no element is valid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+      $scope.Form.email.$setViewValue();
+
+      messageElem = angular.element(element[0].querySelector('#contact > p'));
+      expect(messageElem.hasClass('validation-invalid')).toBe(true);
+    });
+
+    it('should have a success message inside the #contact element when both elements are valid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+      $scope.Form.telephone.$setViewValue('065839481');
+
+      messageElem = angular.element(element[0].querySelector('#contact > p'));
+      expect(messageElem.hasClass('validation-valid')).toBe(true);
+    });
+
+    it('should have a success message inside the #contact element when one of element is valid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+      $scope.Form.telephone.$setViewValue('065839481');
+      $scope.Form.email.$setViewValue();
+
+      messageElem = angular.element(element[0].querySelector('#contact > p'));
+      expect(messageElem.hasClass('validation-valid')).toBe(true);
+    });
+
+    it('should have an error message inside the #contact element when both of elements are invalid', function() {
+      $scope.Form.email.$setViewValue('foo@bar.com');
+      $scope.Form.telephone.$setViewValue('065839481');
+      $scope.Form.email.$setViewValue();
+      $scope.Form.telephone.$setViewValue();
+
+      messageElem = angular.element(element[0].querySelector('#contact > p'));
+      expect(messageElem.hasClass('validation-invalid')).toBe(true);
+    });
+  });
```

Comment:
Me too, I'm thinking about the same thing before. As you can see the current folder of the test, I have combined all the directive spec in different files.

I agree with you that we may get confused afterward, but without making it clear, the test file will harder to modify and fix.

I suggest we create a new test file if given a new test case, for the previous old one we can keep them originally until we enhance them and move them to the new file.

What do you see?